File: NG-Training/NG.NET/TaskTrackerApp/src/app/interfaces/project.ts
export interface Project {
    id: string; // GUID as a string
    projectName: string; // Required field
    description?: string | null; // Optional or nullable
    startDate: Date; // ISO 8601 date-time string (or Date if parsed)
    endDate?: Date | null; // Optional or nullable (ISO 8601 date-time string or Date)
    clientId: string; // GUID as a string
}

File: .NET-Training/API-Training/StandardApiSolution/KnowledgeBase/Models/Project.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace KnowledgeBaseApi.Models
{
    public class Project
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string ProjectName { get; set; }

        public string Description { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime? EndDate { get; set; } // Nullable in case the project is ongoing

        [Required]
        public Guid ClientId { get; set; }
    }
}
